Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -556,8 +556,15 @@ WarpXOpenPMDPlot::WriteOpenPMDParticles (const amrex::Vector<ParticleDiag>& part
   WARPX_PROFILE("WarpXOpenPMDPlot::WriteOpenPMDParticles()");
 
   for (unsigned i = 0, n = particle_diags.size(); i < n; ++i) {
+
     WarpXParticleContainer* pc = particle_diags[i].getParticleContainer();
-    auto tmp = pc->make_alike<amrex::PinnedArenaAllocator>();
+    PinnedMemoryParticleContainer* pinned_pc = particle_diags[i].getPinnedParticleContainer();
```

Comment:
Why is this extra `if` needed now again (can you potentially also add an inline comment)?
Did you move this up from below for clarity?

Reply:
In the previous code, we extracted the name of the particle components to be written from `pc`, e.g. by calling `getParticleRuntimeComps` on `pc` (see a few lines below).

However, there are cases where `pc` (which is a `WarpXParticleContainer`) and `pinned_pc` (which, here, is the buffer that stores the particles that were collected at the boundary) don't have the exact same components. For instance, `pinned_pc` stores `timestamp` (the time at which particles reached the boundary), whereas the corresponding `pc` does not.

For this reason, it is better to call `getParticleRuntimeComps` (and similar functions) on `pinned_pc`, in cases where this buffer is actually used. This is why the `if` condition was added.